src/main/java/com/hm/petmaster/listener/PlayerInteractListener.java

```diff
@@ -140,6 +134,17 @@ public void onPlayerInteractEntityEvent(PlayerInteractEntityEvent event) {
 		}
 	}
 
+	@EventHandler(ignoreCancelled = true)
+    public void onPlayerDamagePet(EntityDamageByEntityEvent event){
+	    if(!((event.getDamager() instanceof Projectile || event.getDamager() instanceof Player) && event.getEntity() instanceof Tameable && ((Tameable) event.getEntity()).getOwner() != null && disablePlayerDamage))
+	        return;
+	    Entity damager = event.getDamager();
+	    if(event.getDamager() instanceof Projectile && ((Projectile) event.getDamager()).getShooter() instanceof Player)
+	        damager = (Player) ((Projectile) event.getDamager()).getShooter();
+	    if(!((Tameable) event.getEntity()).getOwner().equals(damager))
```

Comment:
I don't think this comparison will work properly, you're comparing different object types here. I would suggest using the player names as done elsewhere in the code.